### DISCUSSION TITLE
What data is imported in CSV import and using that functionality

### DISCUSSION THREAD

**Discussion by uriel1998** [bookwyrm-social/bookwyrm#2176] [General]
Exactly what data is being imported and how from the Goodreads import?   I have a lot of indie publisher books that maybe didn't make it into Goodreads, but I have all the appropriate metadata in my Calibre library.

To the point immediately above, the default exported fields in Calibre's CSV export include a description and most of what you need to populate the book fields locally.  (There's even a local path to a cover file, but that'd be a PITA indeed.)  

So if it's as simple as mangling the output CSV from Calibre into the Goodreads format, that's fairly easily doable with external tools. But it appears that if it's not found automagically during the import, the rest of the metadata (such as the book description under the "comments" field in the Calibre export) would be lost.   I have no problem exporting my library to CSV and running another tool on it to populate metadata, but I really don't know what's going to be kept and lost.  Am I just failing my Perception check?

_Originally posted by @uriel1998 in https://github.com/bookwyrm-social/bookwyrm/issues/1698#issuecomment-1112643973_

**Comment by mouse-reeve** [comment]
> Exactly what data is being imported and how from the Goodreads import? I have a lot of indie publisher books that maybe didn't make it into Goodreads, but I have all the appropriate metadata in my Calibre library.

The CSV import just uses identifiers to look up book on external services (Inventaire and Openlibrary) - usually the ISBN or title and author. Metadata in the CSV isn't used to create new books, so even you have columns for things like physical format, publishers, et cetera, those fields aren't used.

> So if it's as simple as mangling the output CSV from Calibre into the Goodreads format, that's fairly easily doable with external tools. But it appears that if it's not found automagically during the import, the rest of the metadata (such as the book description under the "comments" field in the Calibre export) would be lost. I have no problem exporting my library to CSV and running another tool on it to populate metadata, but I really don't know what's going to be kept and lost. Am I just failing my Perception check?

I'm working on a feature to let you create a book from a failed CSV import row, using the metadata in the CSV (the kind of metadata that, as I said above, currently isn't used). This wouldn't happen automatically, but would be part of the troubleshooting phase of the import.  Most of the groundwork is in place for it, but it's still on the to-do list. You can follow the issue here: https://github.com/bookwyrm-social/bookwyrm/issues/1820
